caravel/templates/index.html

```diff
@@ -3,11 +3,43 @@
 {% block head %}
   {{ super() }}
   <script src="/static/js/continuation.js" type="text/javascript"></script>
+  <script src="/static/js/view.js" type=text/javascript></script>
 {% endblock %}
 {% block content %}
-  <div class="row" id="listings">
-    {% for listing in listings %}{% include "listings/inline.html" %}{%
-      endfor %}
-  </div>
+
+  {% for i in range(2) %}
+    {% if loop.first %}
+      <button type="button" class="btn btn-default"><a href="{{ url_for('new_listing') }}" style="text-decoration: none; color: black;">New Listing</a></button>
+    {% else %}
+      <hr>
+      <div class="row" id="listings">
+    {% endif %}
+
+
+    {% if loop.first %}
+    <div class="btn-group" style="float:right;">
+      {% include "thumbnail_button.html" %}
+      {% include "list_button.html" %}
+    </div>
+    {% else  %}
+      {% if view == "Thumbnail" %}
+        {% include "thumbnail_header.html" %}
+      {% else %}
+        <table class="table">
+        {% include "list_header.html" %}
+      {% endif %}
+    {% endif %}
+  {% endfor %}
+
+    {% for listing in listings %}
```

Comment:
Consider this structure:

    {% if view == "thumbnail" %}
       {% for blah blah %}
          ...
       {% endfor %}
    {% else %}
       <table>
          {% for blah blah %}
             ...
          {% endfor %}
       </table>
    {% endif %}